crates/compiler/builtins/bitcode/src/panic.zig

```diff
@@ -2,14 +2,14 @@ const std = @import("std");
 const RocStr = @import("str.zig").RocStr;
 
 // Signals to the host that the program has panicked
-extern fn roc_panic(msg: *const RocStr, tag_id: u32) callconv(.C) void;
+extern fn roc_panic(msg: *const RocStr, tag_id: u32) callconv(.C) noreturn;
 
-pub fn panic_help(msg: []const u8, tag_id: u32) void {
+pub fn panic_help(msg: []const u8, tag_id: u32) noreturn {
     var str = RocStr.init(msg.ptr, msg.len);
     roc_panic(&str, tag_id);
 }
 
 // must export this explicitly because right now it is not used from zig code
-pub fn panic(msg: *const RocStr, alignment: u32) callconv(.C) void {
+pub fn panic(msg: *const RocStr, alignment: u32) callconv(.C) noreturn {
     return roc_panic(msg, alignment);
 }
```

Comment:
I was having a weird error in CI were valgrind was crashing but was not happening locally

My gut feeling is that us adding `unreachable` after a `roc_panic` call is whats causing the issue. 

Looks like unreachable is undefined behavior when compiled with RelaseFast https://ziglang.org/documentation/master/#try which i think we do

This return type lets you do
```
if (some_cond) {
    roc_panic(...)
} else {
    return foo
}
```

without zig type errors

Reply:
So glad this worked. Thanks for the find